OpenRA.Game/ModData.cs

```diff
@@ -162,10 +166,26 @@ public List<MiniYamlNode>[] GetRulesYaml()
 			return Manifest.Rules.Select(s => MiniYaml.FromStream(DefaultFileSystem.Open(s), s, stringPool: stringPool)).ToArray();
 		}
 
+		public SequenceSet GetSequences(IReadOnlyFileSystem fileSystem, string tileset, MiniYaml sequenceDefinitions)
```

Comment:
Reusing the map specific file-system across different maps seems risky - what if two maps ship different sprites but have the same tileset and sequences?

Reply:
IMO we should ignore that edge case.

Hashing all files in the map and checking whether they are used is too expensive IMO, and It should be incredibly rare to have custom assets but with no custom sequences

Reply:
and it's not reusing the filesystem because it's a one time load and done thing

Reply:
though I suppose a modder that is taking full advantage of map directory tracker could encounter it fairly often 🤔

Reply:
Is it possible to have a mod and map distinction here? That would avoid my worry about running into a scenario where we reuse assets between two maps that shouldn't share them. I'm also guessing the mod assets would be a one-off cost we can pay at startup time. Then map assets are presumably small and reloading them for each game would be okay?

Reply:
That was the old loading which #20597 got rid of as too much RAM was in use. It kept the base files always loaded and had map files loaded separately. I dunno if removing this system was the most optimal approach to fixing RAM usage, but we certainly can't just revert back to an identical system

Reply:
I see. The description suggests a map with custom assets causes two copies of the assets in memory. I can see how that's a problem.

I'm still not a fan of crossing our fingers here regarding the filesystems across different maps loading the same data.

Perhaps a different suggestion: If we can detect a map adds no additional assets, then we can preserve the SequenceSet since we know it has only mod assets in it. If we're switching to or from a map with custom assets, we reload the SequenceSet. So when switching between maps that use vanilla mod assets we get the benefit here. Only switching, to, between, or from custom maps requires a reload.

Reply:
The main benefactor should be missions. They do have custom rules, but most or maybe all don't contain custom sequences. The only issue with missions is RA which has a desert shellmap, meaning missions don't get any benefit with this PR. (I mention this in the PR's description)